### ISSUE TITLE
Group Nodes as Iterators

### ISSUE THREAD

**Issue by JacksonMaxfield** [quiltdata/quilt#494]
Currently when I want to loop through a list of nodes contained in a parent node I have to do something like this:

```python
for node_name in pkg.group_node._group_keys():
   actual_node = getattr(pkg.group_node, node_name)
   # do actual work with the node here
```

A nice touch would be to make GroupNode, and I guess PackageNode as well, work like iterators and when I want to loop through something it gives me the node instead of my having to do a roundabout `getattr`.

```python
for node in pkg.group_node:
   # do work
```

**Comment by dimaryaz** [comment]
An iterator sounds like a good idea - but I think it should return all child nodes, not just the ones that are also groups?

That means, the code would look like this:
```
for node in pkg.group_node:
  if isinstance(node, quilt.nodes.GroupNode):
    ...
```

If `isinstance` feels too verbose, we could add some helper functions to nodes - e.g. `_is_group()`?

**Comment by JacksonMaxfield** [comment]
Sorry about my original wording. I agree it should return all child nodes, not just group nodes.

**Comment by JacksonMaxfield** [comment]
@dimaryaz and @akarve on this thread another nice addition would be an `items()` call, depending on the implementation of iterators I just got [this working](https://github.com/AllenCellModeling/QuiltLoader/blob/master/QuiltLoader.py#L128) also as a side note, if the implementations in there look good I can find out where to place them in Quilt Nodes and add them. Just wanted feedback before I toss them in.

**Comment by akarve** [comment]
Resolved in #649.
